service/history/queues/queue_base.go

```diff
@@ -353,6 +369,23 @@ func (p *queueBase) checkpoint() {
 	}
 }
 
+func (p *queueBase) handleAlert(alert *Alert) {
+	// Upon getting an Alert from monitor,
+	// send it to the mitigator for deduping and generating the corresponding Action.
+	// Then run the returned Action to resolve the Alert.
+
+	if alert == nil {
+		return
+	}
+
+	action := p.mitigator.Mitigate(*alert)
```

Comment:
do we need to check alert timestamp? alert may stay in the alertCh for too long that it is no longer relevant? (maybe we should add ability to auto resolve alert. ;)

Reply:
The implementation of the action needs to be able to handle stale/out of date alerts, as checking timestamp is not very reliable. (Also currently we will only have 4 action types, so there will at most 4 items in the channel and unlikely to wait for too long)

For the reader stuck action, if the action can't find any slice containing the stuck range, it won't do anything. 
For other actions like pending task or slice count, if by the time action is running the count is already lower than the threshold, it will be no-op.